commons/src/main/java/fr/free/nrw/commons/upload/ImageProcessing.java

```diff
@@ -0,0 +1,127 @@
+package fr.free.nrw.commons.upload;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.media.ExifInterface;
+import android.net.Uri;
+import android.provider.DocumentsContract;
+import android.provider.MediaStore;
+import android.util.Log;
+
+import java.io.IOException;
+
+
+/**
+ * Created by misao on 16-Dec-15.
+ */
+public class ImageProcessing {
+
+    private Uri uri;
+    private Context context;
+
+    public ImageProcessing(Context context, Uri uri){
+        this.context = context;
+        this.uri = uri;
+    }
+
+    //Gets file path of image from its Uri
```

Comment:
I'm not sure. The Javadocs[0] say null is a possible return value.

[0] http://developer.android.com/reference/android/content/ContentResolver.html#query(android.net.Uri, java.lang.String[], java.lang.String, java.lang.String[], java.lang.String)